tuned/plugins/plugin_disk.py

```diff
@@ -36,13 +37,16 @@ def _init_devices(self):
 
 	def _get_device_objects(self, devices):
 		return [self._hardware_inventory.get_device("block", x) for x in devices]
+	
+	def _is_hdparm_supported(self, device):
+		# Do we want is None here? It returns loop devices that are not supported
+		return device.parent is None or device.parent.subsystem in self._hdparm_subsystems
```

Comment:
I wonder if this is really the way to go to establish `hdparm` support or whether we could use `hdparm`'s exit code to establish support? Disclaimer: haven't looked into that closely.

Reply:
`hdparm` does always return 0 except of invalid inputt (i.e. [https://bugzilla.redhat.com/show_bug.cgi?id=28890](url)). I left this test because of backward compatibility, maybe there would be better to run something like `hdparm -I`and parse output.

Reply:
Interesting,
```
# hdparm -Y /dev/nvme0n1

/dev/nvme0n1:
 issuing sleep command
 HDIO_DRIVE_CMD(sleep) failed: Inappropriate ioctl for device
# echo $?
25
```
seems to work.  I'll look through the code and test this a bit more.

Reply:
Yeah, but it would be nice not to shutdown drivers that support hdparm. I'm looking into `hdparm -C,` if there's possibility to return `unknown` even if the driver supports other commands (probably best option), or `hdparm -i`, but don't want to rely on old driver info.